File: 2588-count-the-number-of-beautiful-subarrays/2588-count-the-number-of-beautiful-subarrays.java
class Solution {
    public long beautifulSubarrays(int[] nums) {
        long count = 0;
        var xorFreq = new HashMap<Integer, Integer>();
        xorFreq.put(0, 1);

        var xor = 0;
        for(var num : nums){
            xor ^= num;
            var freq = xorFreq.getOrDefault(xor, 0);
            count += freq;
            xorFreq.put(xor, 1 + freq);
        }

        return count;
    }
}

File: 2588-count-the-number-of-beautiful-subarrays/2588-count-the-number-of-beautiful-subarrays.py
class Solution:
    def beautifulSubarrays(self, nums: List[int]) -> int:
        count = 0
        xorFreq = {}
        xorFreq[0] = 1

        xor = 0
        for num in nums:
            xor ^= num
            freq = xorFreq.get(xor, 0)
            count += freq
            xorFreq[xor] = 1 + freq

        return count